Update Istio to use OpenMetrics v2 by default (#10304)
DataDog/integrations-core: istio/README.md

Edit the `istio.d/conf.yaml` file (in the `conf.d/` folder at the root of your [Agent's configuration directory][3]) to connect to Istio. See the [sample istio.d/conf.yaml][4] for all available configuration options.

#### Metric collection
To monitor the `istiod` deployment and `istio-proxy` in Istio `v1.5+`, use the following configuration:
    
    ```yaml
    init_config:
    
    instances:

>       - use_openmetrics: true  # This will enable Openmetrics V2 version of the integration

Reviewer: ```suggestion
      - use_openmetrics: true  # Enables Openmetrics V2 version of the integration
```

         - connection_security_policy
    ```
   
**Note**: The `connectionID` Prometheus label is excluded. The [sample istio.d/conf.yaml][8] also has a list of suggested labels to exclude.

   Istio mesh metrics are now only available from `istio-proxy` containers which are supported out-of-the-box via autodiscovery, see [`istio.d/auto_conf.yaml`][9].   

##### OpenMetrics V2 vs OpenMetrics V1
When you enable the `use_openmetrics` configuration option, the Istio integration uses the OpenMetrics V2 implementation of the check. 

In OpenMetrics V2, metrics are submitted more accurately by default and behave closer to Prometheus metric types. For example, Prometheus metrics ending in  `_count` and `_sum` are now submitted as `monotonic_count` by default.

OpenMetrics V2 addresses performance and quality issues in OpenMetrics V1. Updates include native metric types support, improved configuration, and custom metric types.

Set the `use_openmetrics` configuration option to `false` to use the OpenMetrics V1 implementation. To view the configuration parameters for OpenMetrics V1, see [the `conf.yaml.example` file][20].

<div class="alert alert-warning">

> <b>Important Note</b>: If you have multiple instances of Datadog collecting Istio metrics, make sure to use the same implementation of OpenMetrics for all of them. Otherwise, the metrics data will fluctuate in the Datadog app.

Reviewer 2: let's move this to the top of this section, we need users to be aware that all instances should be enabled.

> <b>Important Note</b>: If you have multiple existing instances of Datadog collecting Istio metrics, make sure to use the same implementation of OpenMetrics for all of them. Otherwise, the metrics data will fluctuate in the Datadog app.

Reviewer: ```suggestion
<b>Important Note</b>: If you have multiple instances of Datadog collecting Istio metrics, make sure to use the same implementation of OpenMetrics for all of them. Otherwise, the metrics data will fluctuate in the Datadog app.
```

## Troubleshooting

> ### Invalid chunk length

Reviewer 2: ```suggestion
### Invalid chunk length error
```

> If you see the following error on OpenMetricsV1, update to Python 3 and Agent 7.33.x:
> ```python
> Error: ("Connection broken: InvalidChunkLength(got length b'', 0 bytes read)", InvalidChunkLength(got length b'', 0 bytes read))
> ```

Reviewer 2: ```suggestion
If you see the following error on OpenMetricsBaseCheck (V1) implementation of Istio (Istio integration version `3.13.0` or older):

    ```python
      Error: ("Connection broken: InvalidChunkLength(got length b'', 0 bytes read)", 
      InvalidChunkLength(got length b'', 0 bytes read))
    ```

You can use the Openmetrics V2 implementation of the Istio integration to resolve this error.

Note: you must upgrade to at minimum Agent `7.31.0` and Python 3. See above Configuration section on enabling Openmetrics V2.
```

Why 7.33? This feature is available in 7.31 https://github.com/DataDog/integrations-core/blob/master/istio/CHANGELOG.md#3140--2021-08-22--agent-7310

## Troubleshooting

> ### Invalid Chunk Length

Reviewer: ```suggestion
### Invalid chunk length
```

> If you see the following error on OpenMetricsV1:

Reviewer: ```suggestion
If you see the following error on OpenMetricsV1, update to Python 3 and Agent 7.33.x:
```

```python
Error: ("Connection broken: InvalidChunkLength(got length b'', 0 bytes read)", InvalidChunkLength(got length b'', 0 bytes read))
```

> Please make sure to update to Python 3 and Agent 7.33.x.

Reviewer: ```suggestion
```

## Troubleshooting

### Invalid chunk length error
If you see the following error on OpenMetricsBaseCheck (V1) implementation of Istio (Istio integration version `3.13.0` or older):

    ```python
      Error: ("Connection broken: InvalidChunkLength(got length b'', 0 bytes read)", 
      InvalidChunkLength(got length b'', 0 bytes read))
    ```

You can use the Openmetrics V2 implementation of the Istio integration to resolve this error.

> Note: you must upgrade to at minimum Agent `7.31.0` and Python 3. See above Configuration section on enabling Openmetrics V2.

Reviewer 2: can you link the Configuration section?

```suggestion
Note: you must upgrade to at minimum Agent `7.31.0` and Python 3. See above [Configuration](#configuration) section on enabling Openmetrics V2.
```

that should work for markdown.

Reviewer: ```suggestion
Note: you must upgrade to at minimum Agent `7.31.0` and Python 3. See the [Configuration](#configuration) section on enabling Openmetrics V2.
```
